### PR TITLE
feat: implement new fine tuning job API

### PR DESCRIPTION
**Pull request by henomis**
This PR implements new [Fine Tuning Job API ](https://platform.openai.com/docs/api-reference/fine-tuning).

### CODE REVIEW

**Comment by sashabaranov** [comment]
Hey, thank you so much for coming up with a PR for a new functionality so quickly!

File: fine_tuning_job.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	HasMore bool            `json:"has_more"`
+}
+
+type FineTuningJobEvent struct {
+	Object    string      `json:"object"`
+	ID        string      `json:"id"`
+	CreatedAt int         `json:"created_at"`
+	Level     string      `json:"level"`
+	Message   string      `json:"message"`
+	Data      interface{} `json:"data"`
```
Comment (sashabaranov) [inline fine_tuning_job.go:49]: Can we use `any` instead of `interface{}` here please?

Comment (henomis) [reply fine_tuning_job.go]: Done

File: fine_tuning_job.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+	requestParamters := ""
+	if parameters.after != nil {
+		requestParamters += fmt.Sprintf("after=%s", *parameters.after)
+	}
+	if parameters.limit != nil {
+		if requestParamters != "" {
+			requestParamters += "&"
+		}
+		requestParamters += fmt.Sprintf("limit=%d", *parameters.limit)
```
Comment (sashabaranov) [inline fine_tuning_job.go:136]: Could we please use `url.Values` here?

Example from https://pkg.go.dev/net/url#Values.Encode

```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	v := url.Values{}
	v.Add("cat sounds", "meow")
	v.Add("cat sounds", "mew/")
	v.Add("cat sounds", "mau$")
	fmt.Println(v.Encode())

}
```

Comment (henomis) [reply fine_tuning_job.go]: done

**Review by sashabaranov — APPROVED** [review]

**Pull request merged**